test/fit_tests/common/fit_common.py

```diff
@@ -859,47 +865,61 @@ def apply_obm_settings_seq():
 
 def run_nose(nosepath):
     # this routine runs nosetests from wrapper using path spec 'nosepath'
-    def _noserunner(pathspec):
+    def _noserunner(pathspec, noseopts):
         xmlfile = str(time.time()) + ".xml" # XML report file name
-        return subprocess.call(
-                         [
-                             'export VERBOSITY=' + str(ARGS_LIST['v']) + ';' +
-                             'export ORA=' + str(ARGS_LIST['ora']) + ';' +
-                             'export STACK=' + str(ARGS_LIST['stack']) + ';' +
-                             'export SKU="' + str(ARGS_LIST['sku']) + '";' +
-                             'export NODEID=' + str(ARGS_LIST['nodeid']) + ';' +
-                             'export OBMMAC=' + str(ARGS_LIST['obmmac']) + ';' +
-                             'export VERSION=' + str(ARGS_LIST['version']) + ';' +
-                             'export TEMPLATE=' + str(ARGS_LIST['template']) + ';' +
-                             'export XUNIT=' + str(ARGS_LIST['xunit']) + ';' +
-                             'export GROUP=' + str(ARGS_LIST['group']) + ';' +
-                             'export CONFIG=' + str(ARGS_LIST['config']) + ';' +
-                             'export HTTP=' + str(ARGS_LIST['http']) + ';' +
-                             'export HTTPS=' + str(ARGS_LIST['https']) + ';' +
-                             'export PORT=' + str(ARGS_LIST['port']) + ';' +
-                             'nosetests ' + noseopts + ' --xunit-file ' + xmlfile + ' ' + pathspec
-                         ], shell=True)
+        env = {
+            'VERBOSITY':  str(ARGS_LIST['v']),
+            'ORA':  str(ARGS_LIST['ora']),
+            'STACK':  str(ARGS_LIST['stack']),
+            'SKU': str(ARGS_LIST['sku']) ,
+            'NODEID':  str(ARGS_LIST['nodeid']),
+            'OBMMAC':  str(ARGS_LIST['obmmac']),
+            'VERSION':  str(ARGS_LIST['version']),
+            'TEMPLATE':  str(ARGS_LIST['template']),
+            'XUNIT':  str(ARGS_LIST['xunit']),
+            'GROUP':  str(ARGS_LIST['group']),
+            'CONFIG':  str(ARGS_LIST['config']),
+            'HTTP':  str(ARGS_LIST['http']),
+            'HTTPS':  str(ARGS_LIST['https']),
+            'PORT':  str(ARGS_LIST['port'])
+        }
+        argv = ['nosetests']
+        argv.extend(noseopts)
+        argv.append('--xunit-file')
+        argv.append(xmlfile)
+        argv.extend(pathspec)
+        if nose.run(argv=argv, env=env):
+            return 0
```

Comment:
This is still biting me running the tests, when the subprocess command is removed, it doesn't work.
the variables set via args_list aren't being used...i get "localhost" for the ora.